Added Mass Editing v9

JayVora-SerpentCS opened the pull request.

JayVora-SerpentCS committed:
Added Mass Editing v9

SerpentCS commented:
We must make better index.html

pedrobaeza commented:
Thanks for the PR. We can take the occasion to improve the module to the new standards:
- Put a README.rst file with full OCA template (https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst).
- Put short headers.
- Migrate to new API.
- ...

SerpentCS commented:
Should refer https://github.com/JayVora-SerpentCS/MassEditing
Definitely, we will perform those and make it cleaner. Thanks.

pedrobaeza commented:
Hi, @SerpentCS, do you have time for working in this?

JayVora-SerpentCS commented:
Will do this week.

sudhir-serpentcs committed:
Merge branch '9.0' of https://github.com/OCA/server-tools into 9.0

sudhir-serpentcs committed:
[IMP] Improved the coding standard as per API and PEP8.

sudhir-serpentcs committed:
[ADD] Added README and index file.

sudhir-serpentcs committed:
[IMP] Improved the indentation of the code as per travis test suggestion.

JayVora-SerpentCS commented:
Hi @pedrobaeza ,
Please provide feedback if any.
Thanks,

JayVora-SerpentCS commented:
Hello Reviewers, please update.

pedrobaeza commented on mass_editing/static/description/index.html:
```diff
@@ -0,0 +1,68 @@
+<section class="oe_container oe_dark">
```
This should be included in the README.rst instead of in this HTML field. This way, everyone can see the guide.

JayVora-SerpentCS replied:
In what section of README.rst, we should add the guideline (index.html content)?

pedrobaeza replied:
In the section Usage

pedrobaeza replied:
Remove this file as it's already on the README.rst

JayVora-SerpentCS replied:
We need index.html file for module description on module form view, isnt it?

pedrobaeza replied:
No, the file README.rst now makes this function.

sudhir-serpentcs committed:
Merge branch '9.0' of https://github.com/OCA/server-tools into 9.0

sudhir-serpentcs committed:
[IMP] Improved the README.rst

mdietrichc2c commented:
LGTM :+1: (code review + installation and quick test)

pedrobaeza commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,117 @@
-                raise orm.except_orm(
-                    _("Warning"),
-                    _("Deletion of the action record failed."))
+                raise except_orm(_("Warning"), _("Deletion of the action "
```
Use `UserError` instead of this deprecated exception

JayVora-SerpentCS committed:
Update mass_object.py

JayVora-SerpentCS committed:
Delete index.html

JayVora-SerpentCS commented:
Please check.
Thanks.

moylop260 commented:
Is posible add it test?

sudhir-serpentcs committed:
Merge branch '9.0' of https://github.com/OCA/server-tools into 9.0

sudhir-serpentcs committed:
Merge branch '9.0' of https://github.com/OCA/server-tools into 9.0

sudhir-serpentcs committed:
[ADD] Added unit test cases for MassEditing module.

sudhir-serpentcs committed:
Merge branch '9.0' of https://github.com/JayVora-SerpentCS/server-tools into 9.0

JayVora-SerpentCS commented:
Added unit test cases.
Please check and provide feedback.
Thanks.

cyrilgdn commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,116 @@
+    @api.multi
+    def module_uninstall(self):
+        # search window actions of mass editing  and delete it
+        if self.name == 'mass_editing':
```
Will raised "ValueError: Expected singleton" 
Should start with: 

```
for module in self: 
    if module.name == 'mass_editing':
        [....]
```

JayVora-SerpentCS committed:
Corrected as per @api.multi

JayVora-SerpentCS committed:
Correction for the travis test

JayVora-SerpentCS committed:
Correction for the travis test

JayVora-SerpentCS commented:
@cyrilgdn done

JayVora-SerpentCS commented:
Corrected

carlosdauden commented:
:+1:

lasley commented on mass_editing/__openerp__.py:
```diff
@@ -20,32 +20,25 @@
+    'license': 'GPL-3 or any later version',
+    'summary': 'Mass Editing',
     'depends': ['base'],
+    'images': [],
```
Please remove empty key

lasley commented on mass_editing/__openerp__.py:
```diff
@@ -20,32 +20,25 @@
     ],
-    'installable': False,
+    'installable': True,
     'application': True,
```
Does this fit the definition of an application? Seems more like this module adds functionality to existing applications IMO

lasley commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,116 @@
+                               "created.")
+    field_ids = fields.Many2many('ir.model.fields', 'mass_field_rel',
+                                 'mass_id', 'field_id', 'Fields')
+    ref_ir_act_window = fields.Many2one('ir.actions.act_window',
```
This should be `ref_ir_act_window_id` to fit w/ OCA naming guidelines

lasley commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,116 @@
+                                        help="Sidebar action to make this "
+                                        "template available on records of "
+                                        "the related document model.")
+    ref_ir_value = fields.Many2one('ir.values', 'Sidebar button',
```
This should be `ref_ir_value_id` to fit w/ OCA naming guidelines

lasley commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,116 @@
-                model_ids += found_model_ids
-        return {'value': {'model_ids': [(6, 0, model_ids)]}}
+    @api.onchange('model_id')
+    def onchange_model(self):
```
The original `onchange_model_id` was a more appropriate name for this, as it directly matches the column. Although seems like it should be private too IMO: `_onchange_model_id`

lasley commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,116 @@
-            context)
+        action_obj = self.env['ir.actions.act_window']
+        src_obj = self.model_id.model
+        button_name = _('Mass Editing (%s)') % self.name
```
Looks like we're expecting a singleton record here, but the loop has been removed and there is no `self.ensure_one`

lasley commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,116 @@
+        return super(MassObject, self).unlink()
 
-    def copy(self, cr, uid, record_id, default=None, context=None):
+    @api.one
```
api.one is deprecated

pedrobaeza replied:
`@api.one` has been again "undeprecated". In a nutshell, they have found that it can be useful for @api.constrains methods and compute methods. Anyway, here the use is not correct, as this method returns a value.

lasley replied:
Hahahahahahah I love it, thanks for the elaboration @pedrobaeza - I will keep this in mind for constrains and computations 😄

cyrilgdn replied:
@pedrobaeza Where did you see the "undeprecated" part? Still deprecated in the documentation (https://github.com/odoo/odoo/blob/9.0/openerp/api.py#L405 and in master too https://github.com/odoo/odoo/blob/master/openerp/api.py#L407) 

If it's true, we should change guideline....

pedrobaeza replied:
They commented here https://github.com/odoo/odoo/issues/12408#issuecomment-226421501 the convenience about not deprecating, but they haven't update the documentation yet. We (OCA) also go back with our deprecation warning here: https://github.com/OCA/pylint-odoo/issues/45

lasley commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,116 @@
+        return super(MassObject, self).copy(default)
+
+
+class IrModuleModule(models.Model):
```
Please put all models in separate files per [contribution guidelines](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#file-naming)

lasley commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,116 @@
+    def module_uninstall(self):
+        # search window actions of mass editing  and delete it
+        values_obj = self.env['ir.values']
+        for record in self:
```
This loop can be refactored to `self.filtered(lambda r: r.name == 'mass_editing')` & the if statement removed.

lasley commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,116 @@
+    _inherit = 'ir.module.module'
+
+    @api.multi
+    def module_uninstall(self):
```
Would you mind elaborating on this new method please? Why are the actions not removed with the module during normal uninstall operation?

pedrobaeza replied:
This should use uninstall_hook instead.

lasley commented on mass_editing/__openerp__.py:
```diff
@@ -20,32 +20,25 @@
 #
 ##############################################################################
```
Please change to new OCA header https://github.com/OCA/maintainer-tools/blob/master/template/module/hooks.py#L3

lasley commented on mass_editing/models/ir_model_fields.py:
```diff
@@ -20,26 +20,22 @@
 #
 ##############################################################################
```
Please change to new OCA header https://github.com/OCA/maintainer-tools/blob/master/template/module/hooks.py#L3

lasley commented on mass_editing/models/mass_object.py:
```diff
@@ -20,119 +20,116 @@
 #
 ##############################################################################
```
Please change to new OCA header https://github.com/OCA/maintainer-tools/blob/master/template/module/hooks.py#L3

lasley commented on mass_editing/tests/__init__.py:
```diff
@@ -0,0 +1,23 @@
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+##############################################################################
```
Please change to new OCA header https://github.com/OCA/maintainer-tools/blob/master/template/module/hooks.py#L3

lasley commented on mass_editing/tests/test_mass_editing.py:
```diff
@@ -0,0 +1,92 @@
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+##############################################################################
```
Please change to new OCA header https://github.com/OCA/maintainer-tools/blob/master/template/module/hooks.py#L3

lasley commented on mass_editing/tests/test_mass_editing.py:
```diff
@@ -0,0 +1,92 @@
+        self.mass_object_model = self.env['mass.object']
+        self.res_partner_model = self.env['res.partner']
+        self.partner = self._create_partner()
+        self.partner_model = self.env['ir.model'].\
```
IIRC `self.env['res.partner']._model` would also work, and would skip the DB search

lasley commented on mass_editing/wizard/mass_editing_wizard.py:
```diff
@@ -20,194 +20,256 @@
 #
 ##############################################################################
```
Please change to new OCA header https://github.com/OCA/maintainer-tools/blob/master/template/module/hooks.py#L3

lasley commented:
Thank you for your submission @JayVora-SerpentCS  - Comments inline. IMO I agree w/ Coveralls regarding the need for more tests as well.

JayVora-SerpentCS commented:
Thank you for the review. We will make the necessary changes and update the PR soon.

JayVora-SerpentCS committed:
Merge remote-tracking branch 'upstream/9.0' into 9.0

sudhir-serpentcs committed:
Merge branch '9.0' of https://github.com/JayVora-SerpentCS/server-tools into 9.0

sudhir-serpentcs committed:
[IMP] Improved the code as per review comments, improved copyrights, added test more cases and uninstall hook.

JayVora-SerpentCS commented:
Improved the code as per review comments, added unit test case and uninstall hook.
Please review.

sudhir-serpentcs committed:
[ADD] Added Unit test case for m2m fields.

sudhir-serpentcs committed:
[ADD] Further added unit test case for unlink and multiple fields.

sudhir-serpentcs committed:
[ADD] Added unit test case for model_id onchange.

sudhir-serpentcs committed:
[IMP] Fixed the issue of eval in unit test case.

sudhir-serpentcs committed:
[ADD] Added unit test case for uninstall hook.

sudhir-serpentcs committed:
[ADD] Added unit test case for uninstall hook.

sudhir-serpentcs committed:
[ADD] Added unit test case for wizard's fields_view_get method.

JayVora-SerpentCS commented:
Improved the code as per review comments, added / improved unit test cases and added uninstall hook to unlink the sidebar actions.
@lasley @pedrobaeza Please review.

sudhir-serpentcs committed:
[IMP] Improved the unit test case to cover more lines of code of fields view get.

sudhir-serpentcs committed:
[IMP] Improved the unit test case to cover more lines of code of fields view get.

gurneyalex commented:
:+1:

JayVora-SerpentCS commented:
@lasley @pedrobaeza any feedback on latest changes?

lasley commented:
👍  - can you also squash the commits before merge please?

BT-ojossen commented:
can this be merged?

elicoidal commented:
Yes it can!

pedrobaeza commented:
Well, I would prefer a squashing before merging, because there are a lot of useless commits.

pedrobaeza commented:
But this has to be done by the author, as GitHub only allows to squash all or nothing, but not something in the middle.

elicoidal commented:
@pedrobaeza Well noted

pedrobaeza commented:
Another interesting feature has come to GitHub that allows maintainers to edit branches: https://github.com/blog/2247-improving-collaboration-with-forks, so now we are also able to squash selectively the PRs (but we need to know how to make `git rebase -i HEAD~x`)

JayVora-SerpentCS commented:
Thanks for sharing the knowledge @pedrobaeza. This would help us in future.
